src/main/java/com/chromediopside/service/ProfileService.java

```diff
@@ -137,30 +127,22 @@ private boolean fetchCodeFileUrls(GitHubClient gitHubClient, String username,
       giTinderProfile.setRandomCodeLinks(urlsInString);
       return true;
     } catch (IOException e) {
-      System.out.println(gitHubClientService.getGetRequestIoerror());
+      logService.printLogMessage("ERROR", GitHubClientService.getGetRequestIoerror());
       return false;
     }
   }
 
   public <T> List<T> pickRandomFiveListElements(List<T> list) {
     if(list.size() > 5) {
       Collections.shuffle(list);
-      list = list.subList(0, 5);
+      return list.subList(0, 5);
     }
     return list;
```

Comment:
We should return a list in those cases as well, if the input list contains less than 5 elements, but more than 0.